### PR TITLE
jabref: add xdg_utils to runtime deps

### PR DESCRIPTION
**Pull request by wamserma**
###### Motivation for this change

 see: https://github.com/NixOS/nixpkgs/pull/101287#issuecomment-773768741

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 112028"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wamserma** [comment]
@Coda-Coda @tyrion Can you give this a try? (Bump to 5.2 is postponed due to sth. segfaulting.)

**Comment by dotlambda** [comment]
The bit about Libav is already in https://github.com/NixOS/nixpkgs/pull/111330.

**Comment by wamserma** [comment]
> The bit about Libav is already in #111330.

Right. Fixed this.

File: pkgs/applications/office/jabref/default.nix [hunk 1]
```diff
@@ -24,6 +24,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ makeWrapper wrapGAppsHook ];
   buildInputs = [ gsettings-desktop-schemas ] ++ systemLibs;
+  propagatedBuildInputs = [ xdg_utils ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/jabref/default.nix:27]: Can we reuse the existing wrapper?

Comment (wamserma) [reply pkgs/applications/office/jabref/default.nix:27]: Is there a benefit in using the wrapper? I think this much clearer.

Comment (SuperSandro2000) [reply pkgs/applications/office/jabref/default.nix:27]: I asked on IRC: 

```
18:28:00 <symphorien[m]> <supersandro2000 "propagatedBuildInputs or makeWra"> propagated build inputs only work inside a nix build (or nix-shell)
18:28:15 <symphorien[m]> so only wrapping PATH or patching the source is correct
```

Comment (wamserma) [reply pkgs/applications/office/jabref/default.nix:27]: Ah, good to know. Will update the wrapper.

**Comment by Coda-Coda** [comment]
@wamserma Hi, thanks heaps for the effort. I tried but it doesn't seem to work... I get the same error as before when trying to open a linked PDF.

File: pkgs/applications/office/jabref/default.nix (hunk 1 above)
Comment (wamserma) [reply pkgs/applications/office/jabref/default.nix:27]: Hmm... wrapping PATH does not seem to have any effect for the offending call here: https://github.com/JabRef/jabref/blob/93ad49999badd69c82b052b7d28d4df2fdf24194/src/main/java/org/jabref/gui/desktop/os/Linux.java#L36

And for patching the source we first need to have it build from source...

**Comment by wamserma** [comment]
> @wamserma Hi, thanks heaps for the effort. I tried but it doesn't seem to work... I get the same error as before when trying to open a linked PDF.

@Coda-Coda Sorry, this seems less straightforward than I had hoped for.

**Pull request closed without merging**